lib/domain.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * Domain
+ * @constructor
+ */
+var Domain = function (client) {
+	/**
+	 * Create a new voice call
+	 * @param {Object} params Parameters for creating a new domain
+	 * @param {String} params.name The name is a unique URI to be used in DNS lookups.
+	 * @param {String} params.description String to describe the domain.
+	 * @param {Function} [callback] Callback with the newly created domain
+	 * @return {CallResponse} A promise for the newly created domain
+	 */
+	this.create = function (params, callback) {
+		return client.makeRequest({
+			path   : "domains",
+			method : "POST",
+			body   : params
+		})
+		.then(function (response) {
+			var domain = params;
+			var location = response.headers.location;
+			var domainId = location.substring(location.lastIndexOf("/") + 1);
+			domain.id = domainId;
+			return domain;
+		})
+		.asCallback(callback);
+	};
+
+	/**
+	 * Gets a list of all domains.
+	 * @param {Function} callback A callback with the list of calls
+	 * @return {Promise} A promise for the list of domains.
+	 */
+	this.list = function (callback) {
+		return client.makeRequest({
+			path   : "domains",
+			method : "GET",
+			qs     : { size : 100 }
+		})
+		.then(function (response) {
+			return response.body;
+		})
+		.asCallback(callback);
+	};
+
```

Comment:
Domains api doesn't support pagination (atleast this info is missing in docs). By default GET /domains returns only 25 domains. With param `size` = 100 it returns all domains (max. allowed domains count is 100). Param `page` in query does nothing. So I don't see any sense to support `next here.